filebeat/harvester/reader/multiline.go

```diff
@@ -114,31 +114,34 @@ func (mlr *Multiline) Next() (Message, error) {
 
 func (mlr *Multiline) readFirst() (Message, error) {
 	for {
-		p, err := mlr.reader.Next()
-		if err == nil {
-			if p.Bytes == 0 {
+		message, err := mlr.reader.Next()
+		if err != nil {
+			// no lines buffered -> ignore timeout
+			// TODO: Does this only happen on backoff? Normally is EOF?
+			if err == errMultilineTimeout {
 				continue
 			}
 
-			mlr.startNewLine(p)
-			mlr.state = (*Multiline).readNext
-			return mlr.readNext()
+			// something is wrong here
+			return message, err
 		}
 
-		// no lines buffered -> ignore timeout
-		if err == errMultilineTimeout {
+		// Empty starting lines are skipped
+		if message.Bytes == 0 {
```

Comment:
I was referring to `message.Bytes == 0`. What I try to say is that I'm not sure if we even have to catch this "error". But this kind of disussions should go into https://github.com/elastic/beats/pull/2104 For now I will just remove the comment (no comment) so it is the same as before.